Symfony/CS/Fixer/Contrib/HeaderCommentFixer.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Antonio J. García Lagar <aj@garcialagar.es>
+ */
+class HeaderCommentFixer extends AbstractFixer
+{
+    private static $header = '';
+    private static $headerComment = '';
+
+    /**
+     * Sets the desired header text
+     *
+     * The given text will be trimmed and enclosed into a multiline comment.
+     * If the text is empty, when a file get fixed, the header comment will be
+     * erased.
+     *
+     * @param string $header
+     */
+    public static function setHeader($header)
+    {
+        self::$header = trim((string) $header);
+
+        if (strlen(self::$header) !== 0) {
+            self::$headerComment = self::encloseTextInComment(self::$header);
+        } else {
+            self::$headerComment = '';
+        }
+    }
+
+    /**
+     * @return string
+     */
+    public static function getHeader()
+    {
+        return self::$header;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+
+        if (!count($tokens) || $tokens[0]->getId() !== T_OPEN_TAG) {
+            return $content;
+        }
+
+        $newContent  = $tokens[0]->getContent();
+        $newContent .= PHP_EOL;
+        $newContent .= strlen(self::$headerComment)>0 ? self::$headerComment.PHP_EOL : '';
```

Comment:
Please stop working on content as string.
For now you manipulate what you need and store string result in `$newContent`. Therefore you return it on method output and forgot about `$tokens`. But on 2.x line we changed the interface and we no longer work on code as string, returning it as string etc. We simply update the Tokens collection.
It was done for performance reasons.

Instead of calculating new string representation please update the `$tokens` collection like all other fixers.